source/fundamentals/csfle.txt

```diff
@@ -0,0 +1,224 @@
+
+To run this example, first complete the following steps:
+
+- Save `master-key.txt <https://github.com/mongodb-university/csfle-guides/raw/master/nodejs/master-key.txt>`__
+  to the same directory as this example code.
+- Start a ``mongod`` locally on the default port 27017.
+- Start a ``mongocryptd`` locally on the default port 27020.
+
+.. code-block:: javascript
+  
+   const { MongoClient, Binary } = require("mongodb");
+   const { ClientEncryption } = require("mongodb-client-encryption");
+   const fs = require("fs");
+   
+   async function getRegularClient() {
+     const client = new MongoClient("mongodb://localhost:27017", { 
+       useNewUrlParser: true,
+       useUnifiedTopology: true,
+     });
+     return await client.connect();
+   }
+   
+   async function getCsfleEnabledClient(schemaMap) {
+     const client = new MongoClient("mongodb://localhost:27017", {
+       useNewUrlParser: true,
+       useUnifiedTopology: true,
+       autoEncryption: {
+         keyVaultNamespace: "encryption.__keyVault",
+         kmsProviders: {
+           local: {
+             key: fs.readFileSync("./master-key.txt"),
+           },
+         },
+         schemaMap,
+       },
+     });
+     return await client.connect();
+   }
+   
+   function createJsonSchemaMap(dataKey) {
+     return {
+       "users.ssns": {
+         bsonType: "object",
+         encryptMetadata: {
+           keyId: [new Binary(Buffer.from(dataKey, "base64"), 4)],
+         },
+         properties: {
+           ssn: {
+             encrypt: {
+               bsonType: "int",
+               algorithm: "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic",
+             },
+           },
+         },
+       },
+     };
+   }
+   
+   async function makeDataKey(client) {
+     const encryption = new ClientEncryption(client, {
+       keyVaultNamespace: "encryption.__keyVault",
+       kmsProviders: {
+         local: {
+           key: fs.readFileSync("./master-key.txt"),
+         },
+       },
+     });
+     
+     let dataKey = await encryption.createDataKey("local", {
+       masterKey: null,
+     });
+     
+     return dataKey.toString("base64");
+   }
+   
+   async function run(regularClient, csfleClient) {
+     try {
+   
+       regularClient = await getRegularClient();
+   
+       let dataKey = await makeDataKey(regularClient);
+       console.log(
+         "New dataKey created for this run:\n",
+         dataKey
+       );
+   
+       const schemaMap = createJsonSchemaMap(dataKey);
+       
+       csfleClient = await getCsfleEnabledClient(schemaMap);
+   
+       const regularClientSsnsColl = regularClient
+         .db("users")
+         .collection("ssns");
+       const csfleClientSsnsColl = csfleClient
+         .db("users")
+         .collection("ssns");
+   
+       const exampleDocument = {
+         name: "Jon Doe",
+         ssn: 241014209,
+       };
+   
+       await csfleClientSsnsColl.updateOne(
+         { ssn: exampleDocument["ssn"] },
```

Comment:
```suggestion
         { ssn: exampleDocument.ssn },
```

You might want to run this code through some kind of [formatter](https://prettier.io/playground/) for consistency, accessing an object with a string index is unusual when everything else uses just dot `.` access.

Reply:
Good point, thanks